File: src/test/resources/testpython/code.py

### VIASH START
par = {
  'input': 'input.txt',
  'real_number': 123.987654,
  'whole_number': 17,
  's': '...',
  'truth': True,
  'output': 'output.txt',
  'log': 'log.txt',
  'optional': 'help',
  'optional_with_default': 'me'
}
meta = {
  'resources_dir': '.'
}
### VIASH END

import logging
import sys

if par['log'] is not None:
	logging.basicConfig(filename=par['log'],level=logging.INFO)
else:
	logging.basicConfig(stream=sys.stdout,level=logging.INFO)

logging.info('Parsed input arguments.')

if par['output'] is not None:
    logging.info('Writing output to file')
    output_file = open(par['output'], 'w')
else:
    logging.info('Printing output to console')

def echo(s):
    if par['output'] is not None:
        output_file.write(s + "\n")
    else:
        print(s)

try:
    for key, value in par.items():
        if isinstance(value, list):
            if not value:
                echo(f"{key}: |empty array|")
            else:
                echo(f"{key}: |{','.join(value)}|")
        else:
            echo(f"{key}: |{value}|")

    for key, value in meta.items():
        echo(f"meta_{key}: |{value}|")
finally:
    if par['output'] is not None:
        output_file.close()

File: src/test/resources/testr/code.R
### VIASH START
par <- list(
  input = "input.txt",
  real_number = 123.456,
  whole_number = 789,
  s = "...",
  truth = TRUE,
  output = "output.txt",
  log = "log.txt",
  optional = "help",
  optional_with_default = "me"
)
### VIASH END
par$resources_dir <- resources_dir

write_fun <- function(file, ...) {
  str <- paste0(..., sep = "")
  if (length(file) > 0) {
    write(sub("\n$", "", str), file = file, append = TRUE)
  } else {
    cat(str)
  }
}

write_fun(par$log, "INFO:Parsed input arguments.\n")

str <- paste0(names(par), ": |", par, "|\n", collapse = "")

if (length(par$output) > 0) {
  write_fun(par$log, 'INFO:Writing output to file\n')
} else {
  write_fun(par$log, 'INFO:Printing output to console\n')
}
write_fun(par$output, str)
